app/IOSGateway.m

```diff
@@ -0,0 +1,121 @@
+//
+//  IOSGateway.m
+//  iSH
+//
+//  Created by Miguel Vanhove on 15/02/2019.
+//
+
+#import "IOSGateway.h"
+#import "IACManager.h"
+#import "UIApplication+OpenURL.h"
+
+static IOSGateway *iOSGateway = nil;
+
+@interface IOSGateway ()
+{
+}
+
+@property (copy, nonatomic) NSData *iacResult;
+
+@end
+
+@implementation IOSGateway
+{
+}
+
++ (IOSGateway *_Nullable)sharedSession
+{
+    if (iOSGateway == nil) {
+        iOSGateway = [[IOSGateway alloc] init];
```

Comment:
`[IOSGateway new]` also works, but this is a stylistic choice.

Reply:
Are you java developer ? :) But Ok for the new :)